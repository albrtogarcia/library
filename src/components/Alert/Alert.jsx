import PropTypes from 'prop-types'
import { Icon } from '../Icon/Icon'
import './Alert.scss'

export const Alert = ({ type, title, text }) => {
  let header

  if (title) {
    header = (
      <h5 is className="alert-title">
        {title}
      </h5>
    )
  }
  return (
    <div className={`alert alert-${type}`}>
      <div className="alert-icon">
        <Icon name={type} />
      </div>
      <div className="alert-content">
        {header}
        <p className="alert-message">{text}</p>
      </div>
    </div>
  )
}

Alert.propTypes = {
  type: PropTypes.oneOf(['success', 'warning', 'danger', 'info']),
  title: PropTypes.string,
  text: PropTypes.string
}
