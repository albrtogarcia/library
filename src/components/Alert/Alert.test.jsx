import { describe, expect, it } from 'vitest'
import { Alert } from './Alert.jsx'

describe('Alert', () => {
  const props = {
    type: 'success',
    title: 'Success',
    text: 'This is a success alert'
  }

  it('matches snapshot', () => {
    const alert = Alert(props)
    expect(alert).toMatchSnapshot()
  })
})
