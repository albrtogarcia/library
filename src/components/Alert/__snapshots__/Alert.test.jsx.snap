// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Alert > matches snapshot 1`] = `
<div
  className="alert alert-success"
>
  <div
    className="alert-icon"
  >
    <Icon
      name="success"
    />
  </div>
  <div
    className="alert-content"
  >
    <h5
      className="alert-title"
      is={true}
    >
      Success
    </h5>
    <p
      className="alert-message"
    >
      This is a success alert
    </p>
  </div>
</div>
`;
